GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs

```diff
@@ -644,5 +652,18 @@ private void tsmiRemoveFromList_Click(object sender, EventArgs e)
                 ShowRecentRepositories();
             });
         }
+
+        private void tsmiRemoveMissingReposFromList_Click(object sender, EventArgs e)
+        {
+            RepositoryContextAction(sender as ToolStripMenuItem, repository =>
+            {
+                foreach (Repository repo in _missingRepositories)
+                {
+                    ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.RemoveRecentAsync(repo.Path));
```

Comment:
📝 Operations like `JoinableTaskFactory.Run` can be relatively expensive. It almost certainly won't make a difference at the scale of this method, but it would be possible to avoid some of them by hoisting the operation:

```csharp
ThreadHelper.JoinableTaskFactory.Run(async () =>
{
  foreach (Repository repo in _missingRepositories)
  {
    await RepositoryHistoryManager.Locals.RemoveRecentAsync(repo.Path);
  }
});
```

:bulb: This suggestion only applies to cases where the caller is already on the UI thread, which we know is the case for button click handlers. For other cases, rather than *hoisting* JTF calls, the goal is *removing* them by rewriting the code in a form where they are no longer necessary.

Reply:
Didn't know that, thanks for the tip :)
I fixed it.